src/librustc_typeck/collect.rs

```diff
@@ -2603,6 +2603,35 @@ fn codegen_fn_attrs(tcx: TyCtxt<'_>, id: DefId) -> CodegenFnAttrs {
             }
         } else if attr.check_name(sym::link_name) {
             codegen_fn_attrs.link_name = attr.value_str();
+        } else if attr.check_name(sym::link_ordinal) {
+            use syntax::ast::{Lit, LitIntType, LitKind};
```

Comment:
This branch should be refactored into its own function/method `fn check_link_ordinal(...)`.